### ISSUE TITLE
"Package Installation Error" upon creating a new nUnit test project

### ISSUE THREAD

**Issue by MSK61** [nunit/nunit-vs-templates#21]
I'm trying to create a new nUnit test project from the "New Project" dialog in visual studio 2015. So I select "NUnit 3 Unit Test Project" under Installed->Templates->Visual C#->Test category, and I get this error message:

```
Package Installation Error

Could not add all required packages to the project. The following packages failed to install from 'C:\USERS\MOHAMMED SAFWAT\APPDATA\LOCAL\MICROSOFT\VISUALSTUDIO\14.0\EXTENSIONS\PEI51DEA.4RU\Packages':

NUnit.3.0.1 : Solution is not opened or not saved. Please ensure you have an open and saved solution.
```

I then noticed that in my visual studio options->Projects and Solutions->General I have `Save new projects when created` not checked. If I check this setting and try to create an nUnit test project again like above, the project is created successfully with no error messages.

Honestly I don't know if there's something that can be done here to prevent the offending error message, at least provide clear instructions on what should be done to make it possible to create the project. After all it might be a limitation in visual studio template operability with visual studio settings. I'm submitting this issue just in case it's possible to fix it here.

**Comment by rprouse** [comment]
@MSK61 are your repro steps as follows,
1. Create new solutions
2. Add New Project (Nunit test project)

If so, I expect that just saving your solution with a Save All would fix this. I also expect that it would happen if you add any new project. Is that correct?

**Comment by MSK61** [comment]
In fact my scenario didn't involve creating a solution in a separate step(or having one initially open). My scenario starts with no solutions open at all. I then click "New Project" icon in the toolbar and select the nUnit project template; the error message then ensues(assuming I don't have `Save new projects when created` checked).

**Comment by MSK61** [comment]
I've spent sometime tracking this problem. It's obvious according to https://docs.nuget.org/create/packages-in-visual-studio-templates that the method used in nunit-vs-templates is an official supported approach for adding nuget packages to templates. However it seems that not being able to install nuget packages to non-saved projects is an inherent problem with NuGet.VisualStudio.Interop.dll.

I've found a workaround for that, albeit unofficial and not intended as the typical way to install nuget packages with projects created from templates. We first remove the `<WizardData>` section in _Definitions_project.vstemplate.xml file under the project template(nunit.tests.csharp for c# project). We can then simply add NUnit package through nuget package manager into the project template. This'll create a packages.config file under the project that'll be included in its template zip file, and finally in projects created from this template project. Upon the first build from the project creator, nuget package manager will grab and install the required package from the web repository.

I can create a PR with these changes if it's an acceptable approach. I appreciate feedback from others regarding this idea.

**Comment by ChrisMaddock** [comment]
I imagine the workaround would work, but it would make VS's static analysis pretty unhappy until they user does their first build. Feels a bit like that would create a worse experience for the majority, to fix an issue for the minority.

I'd be tempted to consider this as 'not our bug'. NuGet seem to control both halves of this process (templating, and package restore) - maybe there would be some ground in raising the issue with them? If not, perhaps documenting the problem would be the most sensible solution.

What do you think, @MSK61, @rprouse?

**Comment by ChrisMaddock** [comment]
I've tagged 'confirm' to confirm our plan of action, rather than confirm the bug. I've reproduced.

**Comment by CharliePoole** [comment]
@ChrisMaddock If you are working in this project, it may help to standardize the labels to match framework, console, etc. and create standard ZenHub pipelines.

Usually, for this situation I would use the Discussion pipeline.

**Comment by ChrisMaddock** [comment]
@CharliePoole: I can set up the pipelines. :-)

How would you have labelled a "confirmed bug, but one that we need to decide if we're going to do anything about"? "Idea" didn't really seem right.

**Comment by CharliePoole** [comment]
Based on our earlier emails, where we discussed it in general, it would be is:bug but put it in the Discussion column rather than Backlog. But general discussion is usually not enough. I find I have to try things out for a while and then see how I feel about them. 😄

**Comment by MSK61** [comment]
@ChrisMaddock Of course I do believe this problem very naturally belongs to the way nuget handles dependency packages and the ultimate solution would be to fix the problem there in the first place; this way nothing would have to be done from nUnit template project side.

For the moment I'd be OK with documenting this problem as a limitation until it's(hopefully) fixed with nuget and templates.

**Comment by ChrisMaddock** [comment]
Ok, I've created https://github.com/NuGet/Home/issues/3843. 

Will give it a week or so to see if NuGet accept the bug and apply any priority to it, and then based on the outcome of that, will consider documenting it in the readme here. 😄

**Comment by ChrisMaddock** [comment]
Ok, the issue hasn't been acknowledged by NuGet, but has been referenced by other projects, so this seems to be a general problem.

I think we should close this without action - this doesn't seem to be an NUnit bug, and noting it doesn't really fit in to our documentation. Also. hopefully this issue will now show up via Google if anyone else encounters the same. 

Any objections?

**Comment by rprouse** [comment]
I am okay to close as not our issue.

**Comment by ChrisMaddock** [comment]
Reopening this after stumbling across this solution. Looks like we can add `PromptForSaveOnCreation` to force the user to specify a save location.

Ideally this wouldn't be necessary, but would avoid the obscure error seen here.

https://msdn.microsoft.com/library/twfxayz5.aspx

**Comment by shahnawazm786** [comment]
Hello i am getting the error in visual studio 2013
![image](https://user-images.githubusercontent.com/49604292/99023742-397b5d00-258b-11eb-8120-213d06c95d7c.png)

**Comment by mikkelbu** [comment]
Just for the record it seems that the last issues was solved, see NLog/NLog#2802
